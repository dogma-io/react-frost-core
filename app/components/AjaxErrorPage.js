/**
 * @flow
 */

import {AjaxErrorPage, CodeBlock} from '../../src'
import route from '../factory'
import React, {type Node} from 'react'

const IMPORTS_CODE = "import {AjaxErrorPage} from 'react-frost-core'"

export default route(
  'Ajax error page',
  (): Node => (
    <div>
      <section>
        <CodeBlock code={IMPORTS_CODE} demo={false} language="js" />
      </section>
      <section className="LoadingTypes">
        <CodeBlock language="jsx">
          <AjaxErrorPage
            description="Service is currently unavailable."
            errorCode={500}
            errorDetails="Detailed information would be displayed here."
            errorMessage="Failure to retrieve network elements"
            errorTitle="Internal server error"
            suggestion="Try again later."
          />
        </CodeBlock>
      </section>
    </div>
  ),
)
