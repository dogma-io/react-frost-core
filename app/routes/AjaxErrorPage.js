/**
 * @flow
 */

import route from './factory'
import React, {type Node} from 'react'

export default route('Ajax error page', (): Node => {
  return <div />
})
