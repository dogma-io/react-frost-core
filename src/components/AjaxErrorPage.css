@import '../styles/colors.css';
@import '../styles/typography.css';

:root {
  --frost-color-error-description: var(--frost-color-grey-1);
  --frost-color-error-suggestion: var(--frost-color-grey-1);
  --frost-color-error-http: var(--frost-color-grey-5);
  --frost-color-error-server-message: var(--frost-color-grey-6);
}

.frost-ajax-error-page {
  display: inline-block;
  font-family: var(--frost-font-family);
}

.frost-ajax-error-page-description {
  margin-bottom: 10px;
  color: var(--frost-color-error-description);
  font-size: var(--frost-font-size-xxl);
}

.frost-ajax-error-page-details {
  white-space: pre-line;
}

.frost-ajax-error-page-expand {
  width: 600px;
  max-height: 400px;
}

.frost-ajax-error-page-http-error {
  margin-bottom: 5px;
  color: var(--frost-color-error-http);
  font-size: var(--frost-font-size-m);
}

.frost-ajax-error-page-message {
  margin-bottom: 20px;
  color: var(--frost-color-error-server-message);
  font-size: var(--frost-font-size-s);
}

.frost-ajax-error-page-suggestion {
  margin-bottom: 15px;
  color: var(--frost-color-error-suggestion);
  font-size: var(--frost-font-size-l);
}
