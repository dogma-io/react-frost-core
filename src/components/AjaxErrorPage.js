/**
 * @flow
 */

import Expand from './Expand'
import t from 'grammatic'
import React, {Component, type Node} from 'react'

export type AjaxErrorPageProps = {|
  description: string,
  errorCode: number,
  errorDetails: string,
  errorMessage: string,
  errorTitle: string,
  suggestion?: ?string,
|}

type AjaxErrorPageState = {|
  expanded: boolean,
|}

const COLLAPSED_LABEL = t('Show details', 'Label for toggle in collapsed state')
const EXPANDED_LABEL = t('Hide details', 'Label for toggle in expanded state')
const PREFIX = 'frost-ajax-error-page'

export default class AjaxErrorPage extends Component<
  AjaxErrorPageProps,
  AjaxErrorPageState,
> {
  state = {
    expanded: false,
  }

  _handleExpandChange = (expanded: boolean) => {
    this.setState({expanded})
  }

  render(): Node {
    const {
      description,
      errorCode,
      errorDetails,
      errorMessage,
      errorTitle,
      suggestion,
    } = this.props

    const {expanded} = this.state

    return (
      <div>
        <div className={`${PREFIX}-description`}>{description}</div>
        {suggestion ? (
          <div className={`${PREFIX}-suggestion`}>{suggestion}</div>
        ) : null}
        <div className={`${PREFIX}-http-error`}>
          Error {errorCode} - {errorTitle}
        </div>
        <div className={`${PREFIX}-message`}>{errorMessage}</div>
        <Expand
          className={`${PREFIX}-expand`}
          collapsedLabel={COLLAPSED_LABEL}
          expanded={expanded}
          expandedLabel={EXPANDED_LABEL}
          onChange={this._handleExpandChange}
        >
          <div className={`${PREFIX}-details`}>{errorDetails}</div>
        </Expand>
      </div>
    )
  }
}
