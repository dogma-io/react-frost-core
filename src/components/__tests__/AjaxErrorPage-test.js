import AjaxErrorPage from '../AjaxErrorPage'
import {mount} from 'enzyme'
import React from 'react'

describe('AjaxErrorPage', () => {
  describe('when suggestion provided', () => {
    let wrapper

    beforeEach(() => {
      wrapper = mount(
        <AjaxErrorPage
          description="Service is currently unavailable."
          errorCode={500}
          errorDetails="Detailed information would be displayed here."
          errorMessage="Failure to retrieve network elements"
          errorTitle="Internal server error"
          suggestion="Try again later."
        />,
      )
    })

    it('functions as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('when details expanded', () => {
      let toggle

      beforeEach(() => {
        toggle = wrapper.find('.frost-expand-label-text')
        toggle.simulate('click')
      })

      it('functions as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })

      describe('when details collapsed', () => {
        beforeEach(() => {
          toggle.simulate('click')
        })

        it('functions as expected', () => {
          expect(wrapper).toMatchSnapshot()
        })
      })
    })
  })

  describe('when suggestion omitted', () => {
    let wrapper

    beforeEach(() => {
      wrapper = mount(
        <AjaxErrorPage
          description="Service is currently unavailable."
          errorCode={500}
          errorDetails="Detailed information would be displayed here."
          errorMessage="Failure to retrieve network elements"
          errorTitle="Internal server error"
        />,
      )
    })

    it('functions as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('when details expanded', () => {
      let toggle

      beforeEach(() => {
        toggle = wrapper.find('.frost-expand-label-text')
        toggle.simulate('click')
      })

      it('functions as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })

      describe('when details collapsed', () => {
        beforeEach(() => {
          toggle.simulate('click')
        })

        it('functions as expected', () => {
          expect(wrapper).toMatchSnapshot()
        })
      })
    })
  })
})
