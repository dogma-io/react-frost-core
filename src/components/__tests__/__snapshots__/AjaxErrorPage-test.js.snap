// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AjaxErrorPage when suggestion omitted functions as expected 1`] = `
<AjaxErrorPage
  description="Service is currently unavailable."
  errorCode={500}
  errorDetails="Detailed information would be displayed here."
  errorMessage="Failure to retrieve network elements"
  errorTitle="Internal server error"
>
  <div>
    <div
      className="frost-ajax-error-page-description"
    >
      Service is currently unavailable.
    </div>
    <div
      className="frost-ajax-error-page-http-error"
    >
      Error 
      500
       - 
      Internal server error
    </div>
    <div
      className="frost-ajax-error-page-message"
    >
      Failure to retrieve network elements
    </div>
    <Expand
      className="frost-ajax-error-page-expand"
      collapsedLabel="Show details"
      expanded={false}
      expandedLabel="Hide details"
      onChange={[Function]}
    >
      <div
        className="frost-expand collapsed frost-ajax-error-page-expand"
      >
        <label
          className="frost-expand-label"
          onClick={[Function]}
          role="button"
        >
          <Icon
            icon="chevron"
          >
            <svg
              className="frost-icon frost-icon-frost-chevron"
            >
              <use
                xlinkHref="frost-pack.svg#sprite-chevron"
              />
            </svg>
          </Icon>
          <div
            className="frost-expand-label-text"
          >
            Show details
          </div>
        </label>
      </div>
    </Expand>
  </div>
</AjaxErrorPage>
`;

exports[`AjaxErrorPage when suggestion omitted when details expanded functions as expected 1`] = `
<AjaxErrorPage
  description="Service is currently unavailable."
  errorCode={500}
  errorDetails="Detailed information would be displayed here."
  errorMessage="Failure to retrieve network elements"
  errorTitle="Internal server error"
>
  <div>
    <div
      className="frost-ajax-error-page-description"
    >
      Service is currently unavailable.
    </div>
    <div
      className="frost-ajax-error-page-http-error"
    >
      Error 
      500
       - 
      Internal server error
    </div>
    <div
      className="frost-ajax-error-page-message"
    >
      Failure to retrieve network elements
    </div>
    <Expand
      className="frost-ajax-error-page-expand"
      collapsedLabel="Show details"
      expanded={true}
      expandedLabel="Hide details"
      onChange={[Function]}
    >
      <div
        className="frost-expand expanded frost-ajax-error-page-expand"
      >
        <label
          className="frost-expand-label"
          onClick={[Function]}
          role="button"
        >
          <Icon
            icon="chevron"
          >
            <svg
              className="frost-icon frost-icon-frost-chevron"
            >
              <use
                xlinkHref="frost-pack.svg#sprite-chevron"
              />
            </svg>
          </Icon>
          <div
            className="frost-expand-label-text"
          >
            Hide details
          </div>
        </label>
        <div
          className="frost-expand-content"
        >
          <div
            className="frost-ajax-error-page-details"
          >
            Detailed information would be displayed here.
          </div>
        </div>
      </div>
    </Expand>
  </div>
</AjaxErrorPage>
`;

exports[`AjaxErrorPage when suggestion omitted when details expanded when details collapsed functions as expected 1`] = `
<AjaxErrorPage
  description="Service is currently unavailable."
  errorCode={500}
  errorDetails="Detailed information would be displayed here."
  errorMessage="Failure to retrieve network elements"
  errorTitle="Internal server error"
>
  <div>
    <div
      className="frost-ajax-error-page-description"
    >
      Service is currently unavailable.
    </div>
    <div
      className="frost-ajax-error-page-http-error"
    >
      Error 
      500
       - 
      Internal server error
    </div>
    <div
      className="frost-ajax-error-page-message"
    >
      Failure to retrieve network elements
    </div>
    <Expand
      className="frost-ajax-error-page-expand"
      collapsedLabel="Show details"
      expanded={false}
      expandedLabel="Hide details"
      onChange={[Function]}
    >
      <div
        className="frost-expand collapsed frost-ajax-error-page-expand"
      >
        <label
          className="frost-expand-label"
          onClick={[Function]}
          role="button"
        >
          <Icon
            icon="chevron"
          >
            <svg
              className="frost-icon frost-icon-frost-chevron"
            >
              <use
                xlinkHref="frost-pack.svg#sprite-chevron"
              />
            </svg>
          </Icon>
          <div
            className="frost-expand-label-text"
          >
            Show details
          </div>
        </label>
      </div>
    </Expand>
  </div>
</AjaxErrorPage>
`;

exports[`AjaxErrorPage when suggestion provided functions as expected 1`] = `
<AjaxErrorPage
  description="Service is currently unavailable."
  errorCode={500}
  errorDetails="Detailed information would be displayed here."
  errorMessage="Failure to retrieve network elements"
  errorTitle="Internal server error"
  suggestion="Try again later."
>
  <div>
    <div
      className="frost-ajax-error-page-description"
    >
      Service is currently unavailable.
    </div>
    <div
      className="frost-ajax-error-page-suggestion"
    >
      Try again later.
    </div>
    <div
      className="frost-ajax-error-page-http-error"
    >
      Error 
      500
       - 
      Internal server error
    </div>
    <div
      className="frost-ajax-error-page-message"
    >
      Failure to retrieve network elements
    </div>
    <Expand
      className="frost-ajax-error-page-expand"
      collapsedLabel="Show details"
      expanded={false}
      expandedLabel="Hide details"
      onChange={[Function]}
    >
      <div
        className="frost-expand collapsed frost-ajax-error-page-expand"
      >
        <label
          className="frost-expand-label"
          onClick={[Function]}
          role="button"
        >
          <Icon
            icon="chevron"
          >
            <svg
              className="frost-icon frost-icon-frost-chevron"
            >
              <use
                xlinkHref="frost-pack.svg#sprite-chevron"
              />
            </svg>
          </Icon>
          <div
            className="frost-expand-label-text"
          >
            Show details
          </div>
        </label>
      </div>
    </Expand>
  </div>
</AjaxErrorPage>
`;

exports[`AjaxErrorPage when suggestion provided when details expanded functions as expected 1`] = `
<AjaxErrorPage
  description="Service is currently unavailable."
  errorCode={500}
  errorDetails="Detailed information would be displayed here."
  errorMessage="Failure to retrieve network elements"
  errorTitle="Internal server error"
  suggestion="Try again later."
>
  <div>
    <div
      className="frost-ajax-error-page-description"
    >
      Service is currently unavailable.
    </div>
    <div
      className="frost-ajax-error-page-suggestion"
    >
      Try again later.
    </div>
    <div
      className="frost-ajax-error-page-http-error"
    >
      Error 
      500
       - 
      Internal server error
    </div>
    <div
      className="frost-ajax-error-page-message"
    >
      Failure to retrieve network elements
    </div>
    <Expand
      className="frost-ajax-error-page-expand"
      collapsedLabel="Show details"
      expanded={true}
      expandedLabel="Hide details"
      onChange={[Function]}
    >
      <div
        className="frost-expand expanded frost-ajax-error-page-expand"
      >
        <label
          className="frost-expand-label"
          onClick={[Function]}
          role="button"
        >
          <Icon
            icon="chevron"
          >
            <svg
              className="frost-icon frost-icon-frost-chevron"
            >
              <use
                xlinkHref="frost-pack.svg#sprite-chevron"
              />
            </svg>
          </Icon>
          <div
            className="frost-expand-label-text"
          >
            Hide details
          </div>
        </label>
        <div
          className="frost-expand-content"
        >
          <div
            className="frost-ajax-error-page-details"
          >
            Detailed information would be displayed here.
          </div>
        </div>
      </div>
    </Expand>
  </div>
</AjaxErrorPage>
`;

exports[`AjaxErrorPage when suggestion provided when details expanded when details collapsed functions as expected 1`] = `
<AjaxErrorPage
  description="Service is currently unavailable."
  errorCode={500}
  errorDetails="Detailed information would be displayed here."
  errorMessage="Failure to retrieve network elements"
  errorTitle="Internal server error"
  suggestion="Try again later."
>
  <div>
    <div
      className="frost-ajax-error-page-description"
    >
      Service is currently unavailable.
    </div>
    <div
      className="frost-ajax-error-page-suggestion"
    >
      Try again later.
    </div>
    <div
      className="frost-ajax-error-page-http-error"
    >
      Error 
      500
       - 
      Internal server error
    </div>
    <div
      className="frost-ajax-error-page-message"
    >
      Failure to retrieve network elements
    </div>
    <Expand
      className="frost-ajax-error-page-expand"
      collapsedLabel="Show details"
      expanded={false}
      expandedLabel="Hide details"
      onChange={[Function]}
    >
      <div
        className="frost-expand collapsed frost-ajax-error-page-expand"
      >
        <label
          className="frost-expand-label"
          onClick={[Function]}
          role="button"
        >
          <Icon
            icon="chevron"
          >
            <svg
              className="frost-icon frost-icon-frost-chevron"
            >
              <use
                xlinkHref="frost-pack.svg#sprite-chevron"
              />
            </svg>
          </Icon>
          <div
            className="frost-expand-label-text"
          >
            Show details
          </div>
        </label>
      </div>
    </Expand>
  </div>
</AjaxErrorPage>
`;
